### PR TITLE
Auto rapp installation

### LINKED ISSUE #191: Rapp installer
Develop a tool for installing the dependencies for a given rapp using rosdep.

### PR DESCRIPTION
**Pull request by bit-pirate**
This PR enables the automatic installation of rapps - closes #191.

### CODE REVIEW

**Comment by bit-pirate** [comment]
This PR is ready for review @stonier.

File: rocon_app_manager/launch/standalone_with_capability_server.launch [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <arg name="robot_icon"              value="$(arg robot_icon)" />
+    <arg name="rapp_package_whitelist"  value="$(arg rapp_package_whitelist)" />
+    <arg name="rapp_package_blacklist"  value="$(arg rapp_package_blacklist)" />
+    <arg name="auto_start_rapp"         value="$(arg auto_start_rapp)" />
+    <arg name="screen"                  value="$(arg screen)" />
+    <arg name="capability_server_name"  value="$(arg capability_server_name)"/>
+  </include>
+
+  <!-- *************************** Capability Server *************************** -->
+  <arg name="debug"                                   default="false" />
```
Comment (stonier) [inline rocon_app_manager/launch/standalone_with_capability_server.launch:30]: These would be better up with the rest of the public arguments (took me a while to spot them after trawling through the public arguments first) but it's not a big issue.

File: rocon_app_manager/src/rocon_app_manager/caps_list.py [hunk 2]
```diff
@@ -35,30 +35,31 @@ def __init__(self):
         Sets up a client for the capability server, including a bond.
         Also, retrieves the specifications for the available interfaces and providers from the capability server
 
-        @raise rospy.exceptions.ROSException: Exception is raised when retrieving the capability data
-                                              from the capability server returned errors
-                                              or when waiting for the capability server's services times out
-                                              or when failing to retrieve the capability server's nodelet manager name
+        :raises NotFoundException: Exception is raised when retrieving the capability data
```
Comment (stonier) [inline rocon_app_manager/src/rocon_app_manager/caps_list.py:38]: Another thing for the future that I'm getting in the habit of also: It's useful if you specify the package and exception name (e.g. `rocon_python_comms.NotFoundException`). If you're a user just looking at docs, you won't have the source lying around to go up and infer from whatever imports are there.

File: rocon_app_manager/src/rocon_app_manager/rapp.py [hunk 3]
```diff
@@ -70,6 +71,36 @@ def to_msg(self):
...
+
+        # Trigger the installation of all rapp dependencies
+        rapps = []
+        rapps.append(self.data['name'])
+        try:
+            dependency_checker.install_rapp_dependencies(rapps)
+        except Exception as e:
+            return success, str(e)
+
+        # Update the rospack cache
```
Comment (stonier) [inline rocon_app_manager/src/rocon_app_manager/rapp.py:95]: Be good to have a comment re why the rospack cache needs to be in here (you said something about the pluginlib waiting on future changes for refreshing there).

**Comment by stonier** [comment]
It looks good, though I haven't tested. I'll do that when I merge with Indigo this afternoon and fix errors there.

Just follow up with that comment explaining the rospack cache for me.

Thanks!

**Pull request merged**